src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -196,165 +200,13 @@ export default class SimidMapCreative extends BaseSimidCreative {
-    const placeIcon = {
-      url: this.markerImage_,
-      scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
-    };
-    const placeMarker = new google.maps.Marker({
-      map: this.map_,
-      position: place.geometry.location,
-      icon: placeIcon
-    });
-    ///Recalculate directions if a different active marker is selected.
-    placeMarker.addListener('click', () => {
-      this.activeLocation_ = place.geometry.location;
-      this.displayDirections_();
-    });
-  }
-
-  /**
-   * Displays the directions between the user's current location and current active location.
-   * @private 
-   */
-  displayDirections_() {
-    this.directionsRenderer_.setMap(this.map_);
-    this.calculateRoute_();
-  }
-
-  /**
-   * Creates a drop down menu where users can choose between
-   * different modes of travel to display directions for.
-   * @private 
-   */
-  createTravelChoices_() {
-    const travelChoicesContainer = document.getElementById("button_container")
-    const travelMethod = document.createElement('select');
-    travelMethod.id = "travel_method";
-    TRANSPORT_METHODS.forEach((transportType) =>{
-      const newOption = this.createTravelOption_(transportType);
-      travelMethod.add(newOption);
-    });
-    travelMethod.classList.add("travel_method");
-    travelMethod.addEventListener("change", () => {
-      this.calculateRoute_();
-    });
-    travelChoicesContainer.append(travelMethod);
-  }
-
-  /**
-   * Creates an option element representing a mode of travel.
-   * @param {!string} travelMode A string representing the
-   *   given mode of travel.
-   * @private 
-   */
-  createTravelOption_(travelMode) {
-    const travelOption = document.createElement("option");
-    travelOption.value = travelMode.toUpperCase();
-    travelOption.text = travelMode;
-    return travelOption;
-  }
-
-  /**
-   * Calculates the route between the user's current location and current
-   * active location based off of the selected travel mode.
-   * @private 
-   */
-  calculateRoute_() {
-    const directionsService = new google.maps.DirectionsService();
-    const selectedMode = document.getElementById("travel_method").value;
-    directionsService.route(
-      {
-        origin: this.currentLocation_,
-        destination: this.activeLocation_,
-        travelMode: [selectedMode]
-      },
-      (response, status) => {
-        if (status == "OK") {
-          this.directionsRenderer_.setDirections(response);
-        } else {
-          const directionsErrorMessage = {
-            message: "ERROR: Failed to load directions: " + status,
-          };
-          this.simidProtocol.sendMessage(CreativeMessage.LOG, directionsErrorMessage);
-        }
-      }
-    );
-  }
-
-  /**
+    /**
    * Adds map listeners to the map displayed.
    * @param {!Object} map A Google Maps object.
    * @private 
    */
-  addMapListener_(map) {
+  addMapListener_() {
+    const map = this.simidMap_.getMap();
```

Comment:
I think what is less ideal here, now you need the simid_map_creative to handle the google map object which ideally should just be internal to your SimidMap (bad name) object.

This could be fine but IMO is less clean than pushing this down to the GoogleMapsWrapper object.

Reply:
fixed